src/cursor.c

```diff
@@ -296,6 +289,17 @@ process_cursor_motion(struct server *server, uint32_t time)
 		last_button_hover = LAB_SSD_NONE;
 	}
 
+	if (server->seat.active_view && !server->seat.drag_icon) {
```

Comment:
Could the logic here just be:

```
if (server->seat.active_view && !server->seat.drag_icon) {
  //do stuff
} else if (surface &&
	    !input_inhibit_blocks_surface(&server->seat, surface->resource)) {
  // more stuff
} else {
  wlr_seat_pointer_clear_focus(wlr_seat);  
}
```

Reply:
you are right. added a new fixup commit to simplify the logic